infra/general/PdfGenerator.php

```diff
@@ -124,20 +124,21 @@ public function addTitle($str, PdfStyle $style)
      * @param string $text - the question that should be printed
      * @param PdfStyle $style
      */
-    public function addList($sign, $text, PdfStyle $style)
+    public function addList($sign, $text, PdfStyle $style, $r2lStylePrefix = null)
     {
         if (!isset($text) || !strlen($text) > 0)
         {
             return;
         }
 
+		$r2l = (!empty($r2lStylePrefix) && strpos($style->getStyleName(), $r2lStylePrefix) !== false && $style->getRowIndent() == 'R') ??  false;

```

Comment:
The logic of which style to use must be from the out side, when we are here - the style we got should be the one we should use